### DISCUSSION TITLE
Improving the workflow of translated pages

### DISCUSSION THREAD

**Discussion by NotANormalNerd** [wagtail/wagtail#9231] [Ideas]
Hey everyone,

we are currently reworking some things in our stack and one touchpoint would be to base our customised translation workflow onto the wagtail locale. We have some features that I think would be a great fit for Wagtail and I wanted to have some feedback.
Our workflow is mainly having one translation coordinator working with a agency and the country managers just "QA checking" the final pages before they go live, so it may be that the ideas are a little bit biased in that direction. 

First I want to explain where I see line between Wagtail and wagtail-localize:
- wagtail-localize handles everything regarding the actual translation of text to another language, how to export the text, translate it with whatever software or workflow you have and import it back again into wagtail, as well as the workflow and new states that you need.
- Wagtail itself manages the pages and the page states (live, draft...).

My proposals will be mainly Wagtail itself, as well as maybe better APIs for a wagtail-localize implementation to plug in your own translation workflow.

1. Show all pages in a translation in one View and allow them to be managed (published, bulk published, deleted...)
2. Make PageChoosers translation aware: If I choose a page (chosenPage) on a page and translate that page, the PageChooser should check if there is a chosenPage in the translated language and return that.
3. There should be a way to identify pages in a language that the Editor does not speak: I was thinking about always showing the default locale title below the  current locale title or similar. We had a unique human readable identifier instead of a UUID4 to identify pages across languages.

These are the main points we are currently rethinking on our side and think would be a great fit for Wagtail in general.

Please let me know what you think

**Comment by emilytoppm** [comment]
Thanks for the suggestions - to me they sound great! Tagging @kaedroho and @zerolab as they're more familiar with localize plans

**Comment by lb-** [comment]
Just adding some existing issues / PRs that may work towards these goals, albeit indirectly. 

These are opportunities for contributing, even for older PRs that may need to be rebased / re-submitted.

If these are helpful - please add a comment on them.

- https://github.com/wagtail/wagtail/issues/2719 - ModelAdmin is very powerful and could solve for issue 1 but we don't have bulk actions there yet. It would not be easy but now that we have bulk actions in snippets there is a solid reference of how to do it. 
- https://github.com/wagtail/wagtail/pull/5202 - Supporting Proxy models for Wagtail Page - this PR unfortunately has not been reviewed but could be re-submitted with a rebase. 
- https://github.com/wagtail/wagtail/issues/8480 - this is partially completed and brings some incredible improvements to choosers, including adding a lot of new abilities for customisations. There are still some things to finish off on it but have a read. 
- https://github.com/wagtail/wagtail/pull/9204 - should hopefully make it easier to do more extensive customisations on the Edit view (there are similar other PRs). Might not help much for the idea of users only seeing languages they know but could help. 

Might also be relevant but not particularly key for your called out issues I think. 

- https://github.com/wagtail/wagtail/issues/6918
- https://github.com/wagtail/wagtail/issues/2756

**Reply by zerolab** [reply]
A quick note that #5202 is superseded by #9255 which is pending full review
